core/vm/memory.go

```diff
@@ -59,7 +59,7 @@ func (m *Memory) Set32(offset uint64, val *uint256.Int) {
 	}
 	// Zero the memory area
 	copy(m.store[offset:offset+32], zeroes)
-	val.WriteToSlice(m.store)
+	val.WriteToSlice(m.store[offset:])
```

Comment:
why not `offset:offset+32`?

Reply:
Changed, thanks. WriteToSlice will write 32 bytes but I agree that specifying explicitly reads better, it was just the way I found it.